ring/ring_test.go

```diff
@@ -170,6 +174,104 @@ func TestDoBatchZeroInstances(t *testing.T) {
 	require.Error(t, DoBatch(ctx, Write, &r, keys, callback, cleanup))
 }
 
+func TestDoBatch_QuorumError(t *testing.T) {
+	// we should run several write request to make sure we dont have any race condition on the batchTracker.record code
+	const numberOfOperations = 10000
+	instanceReturnErrors := [3]error{nil, nil, nil}
+	desc := NewDesc()
+	for address := range instanceReturnErrors {
+		instTokens := GenerateTokens(128, nil)
+		instanceID := fmt.Sprintf("%d", address)
+		desc.AddIngester(instanceID, instanceID, "", instTokens, ACTIVE, time.Now())
+	}
+	ringConfig := Config{
+		HeartbeatTimeout:  time.Hour,
+		ReplicationFactor: 3,
+	}
+	ring, err := NewWithStoreClientAndStrategy(ringConfig, "ingester", ringKey, nil, NewDefaultReplicationStrategy(), nil, log.NewNopLogger())
+	require.NoError(t, err)
+	ring.updateRingState(desc)
+	operationKeys := []uint32{1, 10, 100}
+	ctx := context.Background()
+	unfinishedDoBatchCalls := sync.WaitGroup{}
+	runDoBatch := func() error {
+		unfinishedDoBatchCalls.Add(1)
+		returnInstanceError := func(i InstanceDesc, _ []int) error {
+			instanceID, err := strconv.Atoi(i.Addr)
+			require.NoError(t, err)
+			return instanceReturnErrors[instanceID]
+		}
+		return DoBatch(ctx, Write, ring, operationKeys, returnInstanceError, unfinishedDoBatchCalls.Done)
+	}
+
+	// Using 429 just to make sure we are not hitting the &limits
```

Comment:
I figure the ampersand is unintended :)
```suggestion
	// Using 429 just to make sure we are not hitting the limits
```